### PR TITLE
Implemented alltrue, any

### PR DESCRIPTION
**Pull request by MohamedElgharbawy**
Also updated `api_coverage.py` with implemented methods.

### CODE REVIEW

File: tests/numpy/test_misc.py [hunk 1]
```diff
@@ -113,20 +113,55 @@ def test_all(nps_app_inst):
...
+    false_float = nps.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]).reshape(block_shape=(2, 2))
+
+    assert nps.any(true_int)
+    assert not nps.any(false_int)
+    assert nps.any(true_bool)
+    assert not nps.any(false_bool)
+    assert nps.any(true_float).get() == np.any(np.array([[1.0, 0.0, 0.0], [0.0, 0.0, 0.0]]))
+    assert nps.any(false_float).get() == np.any(np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]))
+
     assert nps.all(true_int).dtype is np.bool
```
Comment (elibol) [inline tests/numpy/test_misc.py:136]: Can you put `true_int`, `false_int`, etc. in a list and run these assertions in a loop?

File: tests/numpy/test_misc.py [hunk 2]
```diff
@@ -113,20 +113,55 @@ def test_all(nps_app_inst):
...
+
+    false_int = nps.array([[0, 0, 0], [0, 0, 0]]).reshape(block_shape=(2, 2))
+    false_bool = nps.array([False, False])
+    false_float = nps.array([[0.0, 0.0, 0.0], [0.0, 0.0, 0.0]]).reshape(block_shape=(2, 2))
+
+    assert nps.any(true_int)
+    assert not nps.any(false_int)
+    assert nps.any(true_bool)
+    assert not nps.any(false_bool)
+    assert nps.any(true_float).get() == np.any(np.array([[1.0, 0.0, 0.0], [0.0, 0.0, 0.0]]))
```
Comment (elibol) [inline tests/numpy/test_misc.py:133]: Wherever possible (such as here), construct the numpy array first, and call `nps.array` on it for these kinds of tests. We'll want to make sure we're testing the function's behavior on the same arrays.

**Review by elibol — COMMENTED** [review]
Just left a few comments in the tests.

File: tests/numpy/test_misc.py (hunk 2 above)
Comment (MohamedElgharbawy) [reply tests/numpy/test_misc.py:133]: Definitely. Will do.

File: tests/numpy/test_misc.py (hunk 1 above)
Comment (MohamedElgharbawy) [reply tests/numpy/test_misc.py:136]: Yep, and I'll keep doing this from now on whenever possible.

File: nums/numpy/api.py [hunk 3]
```diff
@@ -382,10 +382,22 @@ def where(condition, x=None, y=None):
     return _instance().where(condition)
 
 
-def all(a: BlockArray, axis=None, out=None, keepdims=False):
+def all(a: BlockArray, axis=None, out=None, keepdims=False, dtype=None):
```
Comment (elibol) [inline nums/numpy/api.py:385]: Can we remove `dtype=None`? It's not a part of the NumPy API (If someone goes from using NumS to NumPy, their code ought to work).

Comment (MohamedElgharbawy) [reply nums/numpy/api.py:385]: Yep, will do.

**Pull request merged**